### DISCUSSION TITLE
migrate typegoose into an Nx scaffold and take advantage of their esmodules standard build feature

### DISCUSSION THREAD

**Discussion by nhhockeyplayer** [typegoose/typegoose#973] [Ideas]
## Describe what you need | want

typegoose can easily avoid most esmodule issues by being buiult into it's own esmodules library off of an Nx scaffold (monorepo)

takes 2 secs to generate

## Do you have already an idea for the implementation?
yes

create nx workspace

create demo app under apps folder using nxconsole IDE plugin

create esmodules compliant TYPEGOOSE library under libs folder using nxconsole IDE plugin

adopt the standard supported testing chassis or continue with original

easy peasy

maintenance of this library would be turnkey overnight and be sponsored by the most elite scaffold on the planet

@NRWL/NX

complete wide open decision space and tons of room to run and thrive forward backward sideways compatibility

**Comment by hasezoey** [comment]
From what i have known until now is that `nx` / nextjs is for front-end / back-end full project use, not for individual libraries unrelated to that ecosystem, though reading the documentation that is also something that they support.
I personally have no experience with `nx` itself and i dont see enough reason to switch to it for this project.

As already outlined in #214, the problem with ESModules is not that it cannot be build anymore, it is that it cannot be tested (without experimental flags) and that there are some inter-op issues that can arise (like ca116198c8e9e5b045ba881558db044c33512481).
